src/net/socket.rs

```diff
@@ -26,51 +31,72 @@ impl SocketSystem {
    }
 
    /// Resolves the socket addresses the given hostname could refer to.
-   async fn resolve_address_with_default_port(hostname: &str) -> anyhow::Result<Vec<SocketAddr>> {
-      if let Ok(addresses) = lookup_host(hostname).await {
-         Ok(addresses.collect())
-      } else {
-         Ok(lookup_host((hostname, relay::DEFAULT_PORT)).await?.collect())
+   async fn resolve_address_with_default_port(hostname: &str) -> anyhow::Result<url::Url> {
+      let mut url = url::Url::parse(&format!("ws://{}", hostname))?;
+
+      if let None = url.port() {
+         // Url::set_port on Error does nothing, so it is fine to ignore it
+         #[allow(unused_must_use)]
+         {
+            url.set_port(Some(relay::DEFAULT_PORT));
+         }
```

Comment:
Please use `let _ = result` for ignoring errors.
```suggestion
         let _ = url.set_port(Some(relay::DEFAULT_PORT));
```